### DISCUSSION TITLE
Calling RTKQ endpoint from a thunk: "tracking" and "unsubscribing"

### DISCUSSION THREAD

**Discussion by azangru** [reduxjs/redux-toolkit#4649] [Q&A] [answered]
I've read the [relevant parts of the documentation](https://redux-toolkit.js.org/rtk-query/api/created-api/endpoints) as best I could; but I am still unclear on how to properly approach the following scenario:

- I want to send a POST request (i.e. perform a mutation) from a thunk (created via `createAsyncThunk`), as a side effect of certain computations.
- I do _not_ want the thunk to wait for the response to that request; but would instead like the thunk to continue synchronously, and to return an action that would be responded to in a reducer.
- I do _not_ need to store the results of the mutation in the slice that RTKQ uses. I do, however, need the RTKQ endpoint to dispatch a fulfilled action upon completion of the mutation, as usual. That fulfilled action will be listened to in the listener middleware.
- I do not want to leak memory 🙂 

So, with all that in mind, I am looking at the `track` option of endpoint's `initiate` method, and at the `unsubscribe` method of the promise returned from that method. The docs say I can set `track` to `false` if I don't need the mutation to persist in RTKQ's slice; and they also say that if I initiate the request manually instead of from a hook, I need to manually unsubscribe from the endpoint; but they do not say whether unsubscribing is still necessary if `track` is set to `false`. Is it?

**Comment by phryneas** [comment, marked as answer]
No, you don't need to call `reset` if you set `track` to false, but it definitely doesn't hurt either.

**Reply by azangru** [reply]
By `reset` you mean `unsubscribe`, right? Like in a code example of a react hook from the docs:

```ts
  useEffect(() => {
    // Add a subscription
    const result = dispatch(api.endpoints.getPost.initiate(postId))

    // Return the `unsubscribe` callback to be called in the `useEffect` cleanup step
    return result.unsubscribe
  }, [dispatch, postId])
```

Or is `reset` a different api?

**Reply by phryneas** [reply]
Mutations have `reset`, queries have `unsubscribe`, I assumed you talked about a mutation here since queries don't have a `track` option.

**Reply by azangru** [reply]
Yes, sorry, you are right, I was talking about a mutation. I did not realise they have a different method from queries.
